### PR TITLE
Fixes for SwiftLint 0.16.1

### PR DESCRIPTION
**Pull request by cocojoe**
Added Lint Pass to Default Lane
Added SwiftLint version check in build phase

### CODE REVIEW

File: Auth0.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1157,7 +1157,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "if which swiftlint >/dev/null; then\nswiftlint\nfi";
+			shellScript = "if which swiftlint >/dev/null && [[ `swiftlint version` =~ \"0.16\" ]] && [[ \"${CONFIGURATION}\" = \"Debug\" ]]; then\nswiftlint\nfi";
```
Comment (hzalaz) [inline Auth0.xcodeproj/project.pbxproj]: If we check the configuration first there is no need to try calling the command.

**Comment by hzalaz** [comment]
Also we need to avoid running it in CIRCLE_CI since time consumed matters

**Comment by cocojoe** [comment]
So I can remove the `lint` pass the `ios ci` build.  However the build script pass will still run, not sure how to get around that.  However in reality we are talking `0m0.134s` to run a lint pass.

**Comment by cocojoe** [comment]
Well actually I can probably check a env to stop build phase script under CI however I still think we should have the official lint lane pass in Circle CI as it helps us catch it when swiftlint gets updated and the run time cost is near 0.

**Pull request merged**